frontend/src/lib/components/LemonCalendar/LemonCalendarRange.tsx

```diff
@@ -0,0 +1,146 @@
+import { LemonCalendar, LemonCalendarProps } from 'lib/components/LemonCalendar/LemonCalendar'
+import React, { useEffect, useState } from 'react'
+import { dayjs } from 'lib/dayjs'
+import { LemonButton } from 'lib/components/LemonButton'
+import { IconClose } from 'lib/components/icons'
+import { formatDateRange } from 'lib/utils'
+
+export interface LemonCalendarRangeProps {
+    value?: (string | null)[]
+    onChange: (date: string[]) => void
+    months?: number
+    getLemonButtonProps?: LemonCalendarProps['getLemonButtonProps']
+    onClose?: () => void
+}
+
+export function LemonCalendarRange({ value, onChange, onClose, months }: LemonCalendarRangeProps): JSX.Element {
```

Comment:
Naming suggestion: To make it more obvious what Ant components this replaces and what the relationship between `LemonCalendarSelect` and `LemonCalendarRange` is, we could name them `LemonDatePicker` and `LemonDatePicker.Range` – WDYT?
The component called `LemonCalendar` _is_ a pure calendar, but it doesn't seem like it should ever be used itself outside of powering date pickers, so I suggest we don't expose it as a `Lemon*` thing – I'm worried someone could try to use and be confused why it's so raw

Reply:
I think this makes sense. The trouble is currently `LemonCalendarRange` and `LemonCalendarSelect` are not really pickers, but just the calendar with apply/cancel buttons and custom handling. They're not `LemonButtonWithPopup`s, but the contents of these popups.

That's because the insight date picker, which is used directly, already makes a popup.

Perhaps when we get around to it, all those date input components could follow the `LemonDatePicker` and `LemonDatePicker.Range` pattern?

Reply:
Fair enough. `LemonDatePicker` would be amazing for forms ASAP though… and with time support, ooh